src/Symfony/Component/DependencyInjection/Compiler/PriorityTaggedServiceTrait.php

```diff
@@ -139,6 +139,10 @@ public static function getDefaultIndex(ContainerBuilder $container, string $serv
 
         $defaultIndex = $rm->invoke(null);
 
+        if (\is_numeric($defaultIndex)) {
```

Comment:
```suggestion
        if (\is_int($defaultIndex)) {
```

I don't think, floats play well here, so let's explicitly check for integers.